src/modules/todo-funtionalities.js

```diff
@@ -0,0 +1,79 @@
+import Storage from './storage.js';
+import Todo from './todo.js';
+
+const toDoList = document.querySelector('.toDoList');
+
+// My To Do list
+let toDoListsArray = [];
+
+const displayTodos = () => {
+  // Get the todos from local storage
+  toDoListsArray = Storage.getDataFromLocalStorage();
+  // Sort the table of todos
+  const sortedTodoLists = toDoListsArray.sort((a, b) => a.index - b.index);
+  let myToDos = '';
+
+  sortedTodoLists.forEach((todo) => {
+    const checked = todo.completed ? 'checked' : '';
+    const complete = todo.completed ? ' completed' : '';
+    myToDos += `
+    <li class="list-item" id="${todo.index}">
+          <article class="status-title">
+            <input data-id="${todo.index}" class="status" type="checkbox" ${checked} title="check!">
+            <input data-id="${todo.index}" type="text" class="toDoEditInput ${complete}" value="${todo.description}" title="Click on '${todo.description}' to EDIT it.">
+          </article>
+          <article class="actions">
+          <span class="material-symbols-outlined remove">delete</span>
+          <span class="material-symbols-outlined verticalDots">more_vert</span>
+          </article>
+          
+          
+        </li>`;
+  });
+  toDoList.innerHTML = myToDos;
+};
+
+const addTodo = () => {
+  const toDoInput = document.querySelector('#toDoInput');
+  const todoDescription = toDoInput.value;
+
+  if (todoDescription.trim().length !== 0) {
+    const todoId = toDoListsArray.length + 1;
+    const todoObj = new Todo(todoDescription, todoId);
+    toDoListsArray.push(todoObj);
+    // Save to local storage
+    Storage.saveToLocalStorage(toDoListsArray);
+    toDoInput.value = '';
+    displayTodos();
+  }
+};
+
+const deleteTodo = (todoId) => {
+  todoId = +todoId;
+  toDoListsArray = toDoListsArray.filter((book) => todoId !== book.index);
+  // Reindexing
+  const arrLength = toDoListsArray.length;
+  for (let i = 0; i < arrLength; i += 1) {
+    toDoListsArray[i].index = i + 1;
+  }
+  // Save to local storage
+  Storage.saveToLocalStorage(toDoListsArray);
+  displayTodos();
+};
+
+const editTodo = (todoId, newVal) => {
+  todoId = +todoId;
+  toDoListsArray = toDoListsArray.map((todo) => {
+    if (todoId === todo.index) {
+      return { ...todo, description: newVal };
+    }
+    return todo;
+  });
+  // Save to local storage
+  Storage.saveToLocalStorage(toDoListsArray);
+  displayTodos();
+};
```

Comment:
The edit todo function fulfills its purpose, however, I kindly suggest implementing some necessary functionalities:
-  When the user stops editing a todo if the todo description is empty you need to make sure that you remove it. Not doing it might result in empty todos as shown in the image below: 

![image](https://user-images.githubusercontent.com/70662720/214098604-d18e803a-25d8-4745-882c-65a0f58cc77b.png)

- The yellow background color is being applied to each todo when the user hovers over it, I kindly suggest that in order to follow the project guidelines you make sure the background color remains there while the user is editing the todo. Otherwise, it might result in a poor user experience. 

![image](https://user-images.githubusercontent.com/70662720/214099475-289de055-f2a4-4050-adfe-b12f78b2b999.png)

![image](https://user-images.githubusercontent.com/70662720/214099623-dae28bf5-94ba-4ddc-8737-736d7e91d17b.png)

- When editing a checked todo, the line-through text decoration shouldn't be showing since it might be difficult to read the description for the user while editing and it could result in a poor user experience.

![image](https://user-images.githubusercontent.com/70662720/214100277-ed95824e-e4b1-4b39-823f-f297eb67c881.png)

- For some reason when editing a checked todo description if you click the checkbox input right after you change its content it will result in some kind of bug returning the todo description to its previous value, I embed a short video for you to see the bug:

![todo list bug](https://media.giphy.com/media/bWjt9eY2joBmiFdi8o/giphy.gif)